models/DataPresentation.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+import {BaseModel} from 'nxus-storage'
+import _ from 'underscore'
+
+var DataPresentation = BaseModel.extend({
+  identity: 'datasets-datapresentation',
+  attributes: {
+    name: 'string',
+    fieldIds: {
+      type: 'array',
+      defaultsTo: []
+    },
+    label: 'string',
+    //data preparation helper
+    extractFieldData: (datasets, datarows) => {
+      let myInstanceObj = this.toObject();
+      return DataPresentation.extractUsingFieldIds(myInstanceObj, datasets, datarows)
+    }
+  }
+});
+
+/**
+ * Extract data relevant to the DataPresentation from collections of DataSet's and DataRow's.
+ * Creates a filtered view object, with summary data about the presentation and all matching DataRows trimmed to just the fields needed.
+ * @param  {[DataPresentation]} presentation 
+ * @param  {[collection of DataSet]} datasets     DataSets that span the field-id's in the supplied presentation. Can be a super-set.
+ * @param  {[collection of DataRow]} datarows     [description]
+ * @return {[Object]}    with properties:
+ *   name - name of the presentation;
+ *   id - ID of the presentation;
+ *   label - label of the presentation;
+ *   fields - array of matching fields extracted from DataSet(s);
+ *   data - array of DataRow records, holding just the identified fields and any primary key fields.
+ */
+DataPresentation.extractUsingFieldIds = (presentation, datasets, datarows) => {

```

Comment:
Is there a reason this is a static class method that takes an instance as the first argument, rather than being an instance method and using `this`?